README.md

```diff
@@ -2,19 +2,137 @@
 
 > Test client for Xo-Server
 
-## Usage
+## Setup
+
+- Xo-server needs to be installed on `localhost:9000`.  
 
-xo-server needs to be installed on `localhost:9000`.
-You should also have a `admin@admin.net` account with password `admin`.
+- You should have a `admin@admin.net` account with password `admin`.
 
-There should be a Xen Server available at `192.168.1.3`with the credentials `root` and password `qwerty`.
+## Adding a test
 
-Warning ! This client will create and delete a lot of items (users, groups, servers ...) to test features
+### Organization
 
 ```
-> npm ci
-> npm test
+src
+├─ user
+|   ├─ __snapshots__
+|   |     └─ index.spec.js.snap 
+|   └─ index.spec.js
+├─ job
+¦   └─ index.spec.js
+¦
+¦
+├─ _xoWithTestHelpers.js
+└─ util.js
 ```
+    
+Let's take an example of `index.spec.js` from `user/`:
+```
+describe("user", () => {
+  describe(".create() :", () => {
+    it("creates a user with permission", async () => {
+      /* some tests */
+    });
+    it("creates a user without permission", async () => {
+      /* some tests */
+    });
+  });
+
+  describe(".delete() :", () => {
+    it("deletes a user successfully with id", async () => {
+      /* some tests */
+    });
+  });
+
+  it("fails trying to create two users, modify them so that they have the same email and connect their session", async () => {
+    /* some tests */
+  });
+});
+```
+
+### Test environment
+
+- The test environment must remain the same before and after each test:  
+  * each resource created must be deleted  
+  * existing resources should not be altered
+
+  There is a helper `_xoWithTestHelpers.js` to create temporary resources and to interface with XO.  
+  Use an array to store the ids of your temporary resources (`userIds` for users,` jobIds` for jobs ...) in order to be able to delete only what you have created.  
+
+  ```
+  import Xo from "xo-lib";
+
+  export default class XoWithTestHelpers extends Xo {
+  constructor(opts) {
+    super(opts);
+    this.userIds = [];
+  }
+
+  async createTempUser(params) {
+    const userId = await super.call("user.create", params);
+    this.userIds.push(userId);
+    return userId;
+  }
+
+  async deleteAllUsers() {
+    await Promise.all(
+      this.userIds.map(id =>
+        super.call("user.delete", { id }).catch(error => console.error(error))
+      )
+    );
+    this.userIds.length = 0;
+  }
+  ```
+  So in your test, use `xo.createTempUser(params)` instead of `xo.call("create.user", params)` to create user.
+
+### Implementation
+
+- Use `_xoWithTestHelpers.js` to create or delete resources.
```

Comment:
We not delete resources using `_xoWithTestHelpers.js`. And i think we can remove this line because it mentioned above.